### ISSUE TITLE
Resolve: URI::InvalidURIError: the scheme http does not accept registry part: errbit.<domain>.com:tcp: (or bad hostname?)

### ISSUE THREAD

**Issue by johnmcconnell** [errbit/errbit#1206]
![snapshot146](https://user-images.githubusercontent.com/1870400/27991800-3121ad30-6437-11e7-8e4e-ef99bc8d462e.png)

**Comment by johnmcconnell** [comment]
This is most likely an affect of the module `URI` from not defaulting to the `http` scheme if none is given.

I haven't confirmed this however.

**Comment by irfanfadilah** [comment]
I also got same issue, the URL become: https://errbit.DOMAIN.com:tcp://IP:PORT/

**Comment by rud** [comment]
How to reproduce the bug? What input, what page? A reproducing test-case would be optimal, as currently this is not easy to act on.

**Comment by stevecrozz** [comment]
I've seen this before when hostnames contain underscores. Some time in the past Ruby's standard library component URI changed and no longer allows hostnames with underscores in them which I believe is more "correct", but some folks use them anyway.

**Comment by rud** [comment]
Right, for DNS underscores are no-go, but I'm sure they're fine on windows networks as machine names

**Comment by encryptblockr** [comment]
why is it only ruby that complains about this??? this is not a problem for many languages for example python...only ruby complains..it hostname works and get routed, why cant ruby to follow along?!?
